### PR TITLE
[misc] Add a minimal example for micro-benchmarks

### PR DESCRIPTION
**Pull request by yolo2themoon**
Related issue = #4058

### CODE REVIEW

File: benchmarks/microbenchmarks/_result.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    def __init__(self):
+        self._items = {'kernel_elapsed_time_ms': kernel_executor}
+
+    @staticmethod
+    def init_taichi(arch: str, result_tag: str):
+        if result_tag == 'kernel_elapsed_time_ms':
+            ti.init(kernel_profiler=True, arch=get_ti_arch(arch))
+        else:
+            return False
+        return True
```
Comment (lin-hitonami) [inline benchmarks/microbenchmarks/_result.py:29]:
```suggestion
        if result_tag == 'kernel_elapsed_time_ms':
            ti.init(kernel_profiler=True, arch=get_ti_arch(arch))
            return True
        return False
```

File: benchmarks/microbenchmarks/_utils.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    for dsize, unit in reversed(size_subsection):
+        if size_in_byte >= dsize:
+            return str(int(size_in_byte / dsize)) + unit
+
+
+def tags2name(tag_list):
+    name = ''
+    for tag in tag_list:
+        name += '_' + tag
+    return name.lstrip("_")
```
Comment (lin-hitonami) [inline benchmarks/microbenchmarks/_utils.py:19]:
```suggestion
    return '_'.join(tag_list)
```

File: benchmarks/microbenchmarks/_items.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    name = 'item'
+
+    def __init__(self):
+        self._items = {}  # {tag: xxx, ...}
+
+    def get(self):
+        return self._items  #dict
+
+    def get_tags(self):
+        return [key for key in self._items]
```
Comment (lin-hitonami) [inline benchmarks/microbenchmarks/_items.py:16]:
```suggestion
        return self._items.keys()
```

File: benchmarks/microbenchmarks/_plan.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            _ms = self._run_func(tag_list)
+            plan['result'] = _ms
+            print(f'{tag_list}={_ms}')
+            ti.reset()
+        rdict = {'results': self.plan, 'info': self.info}
+        return rdict
+
+    def _get_kwargs(self, tags):
+        kwargs = {}
+        for item in self.items:
```
Comment (lin-hitonami) [inline benchmarks/microbenchmarks/_plan.py:45]: If the order of `items` and `tags` are the same, maybe we can iterate through `zip(self.items, tags)` directly?

Comment (yolo2themoon) [reply benchmarks/microbenchmarks/_plan.py]: bf54c75, fixed by removing case_name from the tag list, thanks! 
```
tags = tags[1:] 
for item, tag in zip(self.items, tags):
```

**Review by ailzhang — COMMENTED** [review]
Now I have a basic idea how this works, but will leave the final review to @lin-hitonami :D

File: benchmarks/microbenchmarks/_utils.py (hunk 2 above)
Comment (lin-hitonami) [reply benchmarks/microbenchmarks/_utils.py:19]: `'_'.join(tag_list)` should work exactly as the function here?

Comment (yolo2themoon) [reply benchmarks/microbenchmarks/_utils.py]: my bad, verified locally but missed it when submitting 😭

**Review by lin-hitonami — APPROVED** [review]
LGTM

**Pull request merged**